aiohttp/connector.py

```diff
@@ -1126,3 +1127,53 @@ def path(self) -> str:
             raise ClientConnectorError(req.connection_key, exc) from exc
 
         return cast(ResponseHandler, proto)
+
+
+class NamedPipeConnector(BaseConnector):
+    """Named pipe connector.
+    Only supported by the proactor event loop.
+    See also: https://docs.python.org/3.7/library/asyncio-eventloop.html
+
+    path - Windows named pipe path.
+    keepalive_timeout - (optional) Keep-alive timeout.
+    force_close - Set to True to force close and do reconnect
+        after each request (and between redirects).
+    limit - The total number of simultaneous connections.
+    limit_per_host - Number of simultaneous connections to one host.
+    loop - Optional event loop.
+    """
+
+    def __init__(self, path: str, force_close: bool=False,
```

Comment:
You should validate that it runs under `ProactorEventLoop` somewhere at the beginning of this method, I think.

Reply:
I thought about it, but creating a NamedPipeConnector, raises
`AttributeError: '_WindowsSelectorEventLoop' object has no attribute 'create_pipe_connection'`
Wouldn't that be okay?
But if there is no assertion then the error wouldn't be raised when `loop.create_pipe_connection()` is mocked out. So maybe there is a need for the assertion too.
I'd be fine with whatever seems okay to you.

Reply:
Also, if we end up adding the assert in this case, we'd be adding it here as well, right?
https://github.com/aio-libs/aiohttp/pull/3632/files#diff-dbf24c78287ef0b964d4945f339a43a9R135